src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs

```diff
@@ -52,11 +52,27 @@ public string GetCurrentTestApplicationFullPath()
 
         if (_runtimeFeature.IsDynamicCodeSupported)
         {
+            // Application is running non-native, this is safe to call.
 #pragma warning disable IL3000
             moduleName = Assembly.GetEntryAssembly()?.Location;
 #pragma warning restore IL3000
         }
+        else
+        {
+            bool a = false;
+            if (IsCurrentTestApplicationHostDotnetMuxer && a)
```

Comment:
I think this is fine, I wonder if there's some attribute added to check too, I think that the runtime needs to understand the "mode", I would expected or an attribute or something in the runtime config file, if it's in the attribute we could use that path instead to check the current mutex, but this is a way.

Reply:
@MichalStrehovsky is there an attribute emitted to recognize this scenario?

Reply:
In general I would be comfortable with the assumption that the entry point assembly has a location. It will have it in most cases, but we should not assume so (There's a proposal to allow setting the entry assembly to pretty much anything, and then in theory it could happen even in normal CoreCLR non-single file that the assembly doesn't have a location).